src/Controller.ts

```diff
@@ -39,41 +39,43 @@ export class Controller {
+            return;
+        }
 
-        if (mainClasData) {
-            let targetConfig = this._getLaunchConfig(mainClasData);
-            if (!targetConfig) {
-                targetConfig = await this._createNewLaunchConfig(mainClasData);
-            }
-            app.activeSessionName = targetConfig.name;
-            let jmxport = await getPort();
-            app.jmxPort = jmxport;
-            let vmArgs = '-Dcom.sun.management.jmxremote ' +
-                `-Dcom.sun.management.jmxremote.port=${jmxport} ` +
-                '-Dcom.sun.management.jmxremote.authenticate=false ' +
-                '-Dcom.sun.management.jmxremote.ssl=false ' +
-                '-Djava.rmi.server.hostname=localhost ' +
-                '-Dspring.application.admin.enabled=true';
-            if (targetConfig.vmArgs) {
-                //TODO: smarter merge? What if user is trying to enable jmx themselves on a specific port they choose, for example?
-                vmArgs = vmArgs + ' ' + targetConfig.vmArgs;
-            }
-            const ok: boolean = await vscode.debug.startDebugging(
-                vscode.workspace.getWorkspaceFolder(vscode.Uri.parse(app.path)),
-                Object.assign({}, targetConfig, {
-                    noDebug: !debug,
-                    vmArgs
-                })
-            );
-            if (ok) {
-                // Cannot determine status. It always returns true now.
-                // See: https://github.com/Microsoft/vscode/issues/54214
-            }
-        } else {
-            vscode.window.showWarningMessage("Found no Main Class.");
+        let targetConfig = this._getLaunchConfig(mainClasData);
+        if (!targetConfig) {
+            targetConfig = await this._createNewLaunchConfig(mainClasData);
         }
+        app.activeSessionName = targetConfig.name;
+        let jmxport = await getPort();
+        app.jmxPort = jmxport;
+        let vmArgs = '-Dcom.sun.management.jmxremote ' +
+            `-Dcom.sun.management.jmxremote.port=${jmxport} ` +
+            '-Dcom.sun.management.jmxremote.authenticate=false ' +
+            '-Dcom.sun.management.jmxremote.ssl=false ' +
+            '-Djava.rmi.server.hostname=localhost ' +
+            '-Dspring.application.admin.enabled=true';
+        if (targetConfig.vmArgs) {
+            //TODO: smarter merge? What if user is trying to enable jmx themselves on a specific port they choose, for example?
+            vmArgs = vmArgs + ' ' + targetConfig.vmArgs;
+        }
+        const ok: boolean = await vscode.debug.startDebugging(
+            vscode.workspace.getWorkspaceFolder(vscode.Uri.parse(app.path)),
+            Object.assign({}, targetConfig, {
+                noDebug: !debug,
+                vmArgs
+            })
+        );
+        if (ok) {
+            // Cannot determine status. It always returns true now.
+            // See: https://github.com/Microsoft/vscode/issues/54214
```

Comment:
vscode has fixed this issue? while 
// Cannot determine status. It always returns true now.
            // See: https://github.com/Microsoft/vscode/issues/54214

Reply:
Yes, they have fixed it. But here we don't need this now. We use `onDidStartDebugSession` to determine which session is actually started.